pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -14,7 +14,107 @@
 
 package mixer
 
-// DoNothing holder file.
-func DoNothing() {
-	// do nothing.
+import (
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
+	http_conn "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+	"istio.io/istio/pilot/pkg/proxy/envoy/v1"
+)
+
+// Plugin is a mixer plugin.
+type Plugin struct{}
+
+// OnOutboundListener implements the Callbacks interface method.
+func (m *Plugin) OnOutboundListener(in *plugin.CallbackListenerInputParams, mutable *plugin.CallbackListenerMutableObjects) error {
+	env := in.Env
+	node := in.Node
+	service := in.ServiceInstance.Service
+	proxyInstances := in.ProxyInstances
+
+	if service.MeshExternal {
+		return nil
+	}
+
+	switch in.ListenerType {
+	case plugin.ListenerTypeHTTP:
+		mutable.HTTPFilters = append(mutable.HTTPFilters, buildMixerHTTPFilter(env, node, proxyInstances, true))
+	case plugin.ListenerTypeTCP:
+		mutable.TCPFilters = append(mutable.TCPFilters, buildMixerOutboundTCPFilter(env, node))
+	}
```

Comment:
I think this also should be called only if service.MeshExternal is truye.. In fact, this entire function should be predicated on service.MeshExternal being true.

Reply:
done